pyo3-macros-backend/src/pyfunction.rs

```diff
@@ -186,6 +192,50 @@ pub fn build_py_function(ast: &mut syn::ItemFn, args: PyFunctionAttr) -> syn::Re
     add_fn_to_module(ast, python_name, args)
 }
 
+fn extract_pyo3_metas(attrs: &mut Vec<syn::Attribute>) -> syn::Result<Vec<syn::NestedMeta>> {
+    let mut new_attrs = Vec::new();
+    let mut metas = Vec::new();
+
+    for attr in attrs.drain(..) {
+        if let syn::Meta::List(meta_list) = attr.parse_meta()? {
+            if meta_list.path.is_ident("pyo3") {
+                for meta in meta_list.nested {
+                    metas.push(meta);
+                }
+            } else {
+                new_attrs.push(attr)
+            }
+        }
+    }
+    attrs.extend(new_attrs);
```

Comment:
After the drain, `attrs` will be empty, so it's more efficient to just overwrite.

```suggestion
    *attrs = new_attrs;
```

Reply:
Fixed, thanks!